### ISSUE TITLE
Improve tutorial on sources

### ISSUE THREAD

**Issue by Matrim-Cauthon** [xjsframework/xjsframework.github.io#23]
I'm talking about this link http://xjsframework.github.io/tutorials.html#/sourceplugins

I know we modified a huge chunk of the tutorial because of the global sources feature, but at the moment, I think there is still a lot we can improve on. From a first-time developer's perspective, there are a lot of areas in the tutorial that don't make a lot of sense. Let's get our heads together and restructure the flow of the page.

A few checkpoints we can revisit:
- the Rendering Properties table is located at the bottom of the page, with no context describing it. We need to put it closer to where we describe source properties versus renderer (videoitem) properties
- in the same table I noticed we still use the `Source` class, but I believe that name and custom name are now renderer properties instead of source properties
- let's put all migration notes into their own boxes and label them as "Migrating from xjs@1.3.0 to newer versions", so that new developers can opt to skip those areas and not get distracted.
- let's add more screenshots! Pictures tell a thousand words. For example, we can add a screenshot of how to add a source. We can add an illustration (perhaps just file a graphics request from the graphics team) that helps clarify what a global source is, and/or the split between the source and the videoitem.

I'll put this into milestone 1.5.0 as I think fixing this is a huge priority.

**Comment by Matrim-Cauthon** [comment]
Assigning to @SML-MeSo for now, but just feel free to reassign this to anyone.

**Comment by Matrim-Cauthon** [comment]
@SML-MeSo @virn I pushed a bunch of modifications to the tutorials, latest commit at 32cacd0.

There are a number of big changes there that are also related to xjsframework/xjs#111 --- you might want to take a thorough look at the docs. My original intention was to make the docs actually prescriptive of the direction we should take for 1.5.0, especially with how we are tackling the issue of a cleaner API for sources and items.

I also peppered a large number of `TODO`s in the tutorials; let's make sure to handle them all before release of the next version.
